docs/authProtocolExplainer.md

```diff
@@ -0,0 +1,163 @@
+
+B. Generate Authentication Request.
+: The OpenID Connect Self-Issued Authentication Request with DID signatures is formed and signed.
+
+C. Application Response with Authentication Request.
+
+:  The webpage loads with a user action available (such as a button) that allows the user to sign into the application using their DID. The Authentication Request that the webpage will send to the User Agent is formed serverside and is included in this response back to the browser (See [OpenID Connect Claim Parameters](https://openid.net/specs/openid-connect-core-1_0.html#Claims) for more information about what types of claims are allowed).
+
+: *An Authentication Request Object.*
+This object will be put in a JWS signed by the Application.
+```=JSON
+/**
+ * OpenID Connect Self-Issued Authentication Request
+ */
+{
+  /** DID of the issuer of the request.
+   * This should match the signature. 
+   */
+  "iss": "did:doc:application",
+  /** MUST be set as 'id_token' in order to match OIDC self-issued protocol */
+  "response_type": "id_token",
+  /** The redirect url as specified in the OIDC self-issued protocol */
+  "client_id": "app://claims",
+  /** MUST be set to "openid" */
+  "scope": "openid",
+  /** Opaque value used by issuer for state */
+  "state": af0ifjsldkj,
+  /** Request Nonce */
+  "nonce": "drnEJZTtfh",
+  /** 
+   * Claims that are requested.
+   */
+  "claims":{
+            "id_token":{
+                "age": {"essential": true}
+            }
+        }
+}
+```
+
+D. DID Authentication Request.
+
+:  The Authentication Request is sent to the User Agent in various ways:
+   - In the desktop browser-browser extension flow, the Authentication Request is sent by a customized navigator.did.requestAuthentication function.
+   - In the desktop browser-mobile device flow, the Authentication Request is sent by creating a QR code and scanning the QR code using the camera on a mobile device.
+   - In the mobile browser-mobile device flow, the Authentication Request is sent by deep linking to the UA from the mobile browser.
+
+E. DID Query for Application's DID.
+:  The User Agent queries for the application's DID document using a Universal Resolver.
+
+F. DID Resolver Result for Application's DID.
+:  The Universal Resolver sends back the DID Document of the queried DID. This DID Document contains information about the DID including it's public key.
+
+G. Application's DID Signature Verification.
+:  The User Agent uses the applicationn's public key obtained by Step E to verify the signature on the JWT.
+
+    > A note about Steps E-G: These steps make up a security measure to protect against malicious applications posing as real businesses. The Authentication Request will be signed using the application's DID in the form of a JWT, and the User Agent will verify this signature in order to confirm that this application is not pretending to be another.
```

Comment:
unindent? This should appear as 
> Quote

but its showing as

       > Code

Reply:
changed formatting.